app/src/test/java/org/grammatek/simacorrect/spellcheckerservice/YfirlesturAnnotationTest.kt

```diff
@@ -0,0 +1,219 @@
+            val annotations = ylAnnotations.suggestionsIndices
+
+            assertThat(annotations.size).isEqualTo(expectedAnnotations[i].size)
+            for (j in annotations.indices) {
+                assertThat(annotations[j].startChar).isEqualTo(expectedAnnotations[i][j].startChar)
+                assertThat(annotations[j].endChar).isEqualTo(expectedAnnotations[i][j].endChar)
+            }
+        }
+    }
+
+    @Test
+    fun `does Yfirlestur token count match our constructed token count`() {
+        val suggestionLimit = 5
+        val dictionary = arrayListOf<String>()
+        for (testCase in testCases) {
+            val response = api.correctApiPost(testCase)
+            val ylAnnotation = YfirlesturAnnotation(response, testCase)
+            ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+            var size = 0
+            for (tokensIndex in ylAnnotation.tokensIndices) {
+                size += tokensIndex.size
+            }
+            assertThat(size).isEqualTo(response.stats?.numTokens ?: 0)
+        }
+    }
+
+    @Test
+    fun `is dictionary taken into account for suggestions`() {
+        val suggestionLimit = 5
+        val text = "Eg dreymi"
+        val response = api.correctApiPost(text)
+        val dictionary = arrayListOf<String>()
+        var suggestionsInfo = YfirlesturAnnotation(response, text).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        assertThat(suggestionsInfo).isNotEmpty()
+
+        dictionary.add("Eg")
+        suggestionsInfo = YfirlesturAnnotation(response, text).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        assertThat(suggestionsInfo).isEmpty()
+
+        dictionary.clear()
+        suggestionsInfo = YfirlesturAnnotation(response, text).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        assertThat(suggestionsInfo).isNotEmpty()
+    }
+
+    @Test
+    fun `is suggestion limit enforced`() {
+        var suggestionLimit = 0
+        val text = "eg dreimi um að leita af mindinni"
+        val response = api.correctApiPost(text)
+        val dictionary = arrayListOf<String>()
+
+        val ylAnnotation = YfirlesturAnnotation(response, text)
+        var suggestionInfo = ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        for (suggestions in suggestionInfo) {
+            assertThat(suggestions.suggestionsCount).isAtMost(suggestionLimit)
+        }
+
+        suggestionLimit = 1
+        suggestionInfo = ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        for (suggestions in suggestionInfo) {
+            assertThat(suggestions.suggestionsCount).isAtMost(suggestionLimit)
+        }
+    }
+
+    @Test
+    fun `multiple whitespaces don't throw off indices for annotations`() {
+        val suggestionLimit = 5
+        val text = "   Afhverju  er   það  eki  hægt"
```

Comment:
What about trailing whitespaces, tabs, emojis at the end or before a trailing whitespace or after a leading whitespace ?

Reply:
Trailing whitespaces aren't really any concern since they are basically ignore (as they should).
Good point with tabs, hadn't really taken that into account since its on phone but it could be from copied text or something. 
Adding a test case with emojis would be useful, especially since their slightly different to typical characters in terms of unicode code points